keps/sig-architecture/3935-oldest-node-newest-control-plane/README.md

```diff
@@ -0,0 +1,799 @@
+    Qualify v1.43                                     : n2_143qual,        2031-04-01,4w
+    Control Plane → v1.41 → v1.42                     : n2_142cpupgrade,   after n2_143qual, 1d
+    Nodes → v1.42 (drain/recreate workloads)          : n2_142nodeupgrade, after n2_142cpupgrade,  1w
+    Control Plane → v1.43                             : n2_143cpupgrade,   after n2_142nodeupgrade,  1d
+    Nodes → v1.43 (drain/recreate workloads)          : n2_143nodeupgrade, after n2_143cpupgrade,  1w
+    Run v1.43...                                      : n2_143run,         after n2_143nodeupgrade, 2031-07-01
+```
+-->
+
+There is nothing magical about the number 2 in the existing version skew policy.
+That policy intended to allow the oldest supported node to work with the newest supported control plane (exactly what this KEP is trying to accomplish).
+When there are 3 supported minor versions, max skew of n-2 appears to make sense.
+
+However, as part of the [yearly support period](https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/1498-kubernetes-yearly-support-period#proposal) KEP,
+it became clear that overlap between the oldest and newest version is required to actually give users a version they can use for a year.
+This led to the current policy of supporting each minor version for approximately 14 months, giving users a 2-month period to qualify and upgrade following a new Kubernetes release
+before the oldest supported minor version reaches end-of-life. During that 2-month period, there are actually 4 supported minor versions.
+
+Updating the skew policy to allow up to n-3 nodes brings the policy back in line with its original goals.
+
+### Goals
+
+* Allow users to make full use of the [yearly support period](https://github.com/kubernetes/enhancements/tree/master/keps/sig-release/1498-kubernetes-yearly-support-period)
+* Streamline upgrade paths for nodes / node pools to minimize required workload churn
+* Validate oldest supported node components work with latest control plane components
+* Update the [Kubernetes version skew policy](https://kubernetes.io/releases/version-skew-policy/) to indicate node components from the oldest supported minor version work with control plane components from the newest supported minor version, starting with Kubernetes `v1.25` nodes and `v1.28` control planes.
+
+### Non-Goals
+
+* This does not propose any changes to the number of supported minor versions
+* This does not propose any changes to the minor version support duration
+* This does not propose any changes to the Kubernetes release process
+* This does not propose any changes to the mechanics control plane components use to enable features 
+  and work with older nodes without features ([KEP-3920](https://github.com/kubernetes/enhancements/pull/3920) could help with this)
+* This does not propose delaying introduction of existing features SIGs have planned around existing n-2 node/control plane skew
```

Comment:
But the new features will be introduced slower, correct? A new api fields progression thru stages. Today we have alpha (release:n), beta (release:n+1), stable (release:n+3), feature gate removed (release:n+5). Now we will have stable (release:n+4), feature gate removed (release:n+7)?

Reply:
found comment about it down below. So this is just talking about in-flight features than

Reply:
> Today we have alpha (release:n), beta (release:n+1), stable (release:n+3), feature gate removed (release:n+5)

Some features graduate to GA in n+2, it's actually rare for us to wait for a feature to reach the oldest nodes before graduating it. I highlighted the features over the last couple years where we _did_ wait for the oldest supported nodes before graduating, and it was typically security-related things and cleanups that had to wait.

> Now we will have stable (release:n+4), feature gate removed (release:n+7)?

Changes that wait for the oldest support node before graduating would have to wait one more release, but removal of feature gate flags isn't tied to skewed nodes, we just try to give a couple releases for folks to clean up inert feature gates from their invocations